### PR TITLE
horizontalAlign = center, verticalAlign = middle

### LINKED ISSUE #43: horizontalAlign = center, verticalAlign = middle
Feature request: "center" should be another valid value for `horizontalAlign`, likewise "middle" for `verticalAlign`

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #43, https://github.com/PolymerElements/paper-dialog/issues/124, https://github.com/PolymerElements/paper-toast/issues/83

### CODE REVIEW

File: iron-fit-behavior.html [hunk 1]
```diff
@@ -600,23 +639,24 @@
...
-        // Check which crops less. If it crops equally, check if align is ok.
-        if (diff < 0 || (diff === 0 && alignOk)) {
-          position = pos;
-        }
         // If not cropped and respects the align requirements, keep it.
         // This allows to prefer positions overlapping horizontally over the
         // ones overlapping vertically.
-        if (position.croppedArea === 0 && alignOk) {
+        if (pos.croppedArea === 0 && alignOk) {
+          position = pos;
```
Comment (e111077) [inline iron-fit-behavior.html]: I have to say `position` and `pos` are pretty bad names. Maybe something along the lines of `finalPosition` || `outPosition` || `acualPosition` and `currentPosition

File: iron-fit-behavior.html [hunk 2]
```diff
@@ -600,23 +639,24 @@
...
           break;
         }
+        position = position || pos;
+        var diff = pos.croppedArea - position.croppedArea;
+        // Check which crops less. If it crops equally, check if at least one align setting is ok.
+        if (diff < 0 || (diff === 0 && (pos.verticalAlign === vAlign || pos.horizontalAlign === hAlign))) {
+          position = pos;
+        }
```
Comment (e111077) [inline iron-fit-behavior.html]: Why were these lines moved below the if statement? Does this fix another issue?

File: iron-fit-behavior.html (hunk 1 above)
Comment (e111077) [reply iron-fit-behavior.html]: Also, doesn't this section break API? position will always return as pos instead of `position = position || pos` which was moved below this if statement

Comment (valdrinkoshi) [reply iron-fit-behavior.html]: Yeah, I agree that the naming is not great :x Will update.

Regarding the change: `alignOk` is computed from `pos`. Previously, we were looking `position.croppedArea` instead of `pos.croppedArea`, which was basically like saying `positions[1] has the correct align properties and positions[0] doesn't crop. Let's pick positions[0]!`.

In this change I wanted to make it clear that if the current position doesn't crop and has the correct alignments, we use that. Otherwise, we use search for the best position that doesn't crop (see lines below...)

File: iron-fit-behavior.html (hunk 2 above)
Comment (valdrinkoshi) [reply iron-fit-behavior.html]: Mainly for clarity: we need to compute the diff cropped area only if we didn't find the "perfect match". The difference from the previous implementation is that we don't check for both vertical and horizontal alignment to match, but rather one or the other.

**Comment by e111077** [comment]
explained offline
